apps/fxc-front/src/app/logic/score/improvedScorer.ts

```diff
@@ -0,0 +1,231 @@
+import { Point } from '@flyxc/common';
+import { Point as XcScorePoint, scoringRules, Solution } from 'igc-xc-score';
+import { BRecord, IGCFile } from 'igc-parser';
+import { CircuitType, Score } from './scorer';
+import { LeagueCode } from './league';
+
+async function lazyLoadedSolver(): Promise<typeof _solver> {
+  if (!_solver) {
+    const { solver } = await import('igc-xc-score');
+    _solver = solver;
+  }
+  return _solver;
+}
+
+let _solver: (
+  flight: IGCFile,
+  scoringRules: object,
+  config?: { [key: string]: any } | undefined,
+) => Iterator<Solution, Solution>;
+
+export type ScoreAndRoute = { score: Score; route: Point[] };
+export type ScoringTrack = { lat: number[]; lon: number[]; timeSec: number[]; minTimeSec: number };
+
+export async function scoreTrack(track: ScoringTrack, leagueCode: LeagueCode): Promise<ScoreAndRoute | undefined> {
+  const scoringRules = getScoringRules(leagueCode);
+  if (scoringRules) {
+    const solver = await lazyLoadedSolver();
+    const solutions = solver(igcFile(track), scoringRules, undefined);
+    const solution = solutions.next().value;
+    return { score: toScore(solution), route: toRoute(solution) };
+  }
+  return undefined;
+}
+
+function toScore(solution: Solution): Score {
+  return new Score({
+    distance: solution.scoreInfo?.distance,
+    points: solution.score,
+    multiplier: solution.opt.scoring.multiplier,
+    circuit: toCircuitType(solution.opt.scoring.code as CircuitTypeCode),
+    indexes: getIndexes(solution),
+    closingRadius: solution.scoreInfo?.cp?.d,
+  });
+}
+
+// duplicated code from apps/fxc-front/src/app/logic/track.ts
+type CircuitTypeCode = 'od' | 'tri' | 'fai' | 'oar';
+
+function toCircuitType(code: CircuitTypeCode) {
+  switch (code) {
+    case 'od':
+      return CircuitType.OpenDistance;
+    case 'fai':
+      return CircuitType.FaiTriangle;
+    case 'oar':
+      return CircuitType.OutAndReturn;
+    case 'tri':
+      return CircuitType.FlatTriangle;
+  }
+}
+
+// end duplication
+
+function getIndexes(solution: Solution) {
+  let currentIndex = -1;
+  const entryPointsStart = getEntryPointsStart(solution);
+  const result = entryPointsStart ? [currentIndex++] : [];
+  const closingPointsIn = getClosingPointsIn(solution);
+  if (closingPointsIn) result.push(currentIndex++);
```

Comment:
style: flyxc uses {} even for single line blocks - isn't that enforced by the formatter?

Reply:
fixed